src/storage/standard_buffer_manager.cpp

```diff
@@ -329,12 +329,13 @@ struct BlockIndexManager {
 };
 
 class TemporaryFileHandle {
-	constexpr static idx_t MAX_ALLOWED_INDEX = 4000;
+	constexpr static idx_t MAX_ALLOWED_INDEX_BASE = 4000;
 
 public:
-	TemporaryFileHandle(DatabaseInstance &db, const string &temp_directory, idx_t index)
-	    : db(db), file_index(index), path(FileSystem::GetFileSystem(db).JoinPath(
-	                                     temp_directory, "duckdb_temp_storage-" + to_string(index) + ".tmp")) {
+	TemporaryFileHandle(idx_t temp_file_count, DatabaseInstance &db, const string &temp_directory, idx_t index)
+	    : max_allowed_index((1 << temp_file_count) * MAX_ALLOWED_INDEX_BASE), db(db), file_index(index),
```

Comment:
```suggestion
	    : max_allowed_index((1ull << temp_file_count) * MAX_ALLOWED_INDEX_BASE), db(db), file_index(index),
```
`1 << temp_file_count` is computed as a 32 bit integer, so just to stay safe I would go explicitly for using 64bit types.

Reply:
I agree that this should have been explicitly typed. However, it will never be an issue because when `temp_file_count` is 30 (right before it overflows), we already have a temp file that can store more than 1 exabyte.

Reply:
Considering `temp_file_count` is an `idx_t`, will it actually be computed as a 32 bit integer?

Reply:
I tried this:
```c++
#include <iostream>
 
int main()
{
    auto one = 1;
    auto bigone = 1ull;
    uint64_t exp = 32;
    std::cout << (one << exp)<< '\n'
              << (bigone << exp)<< '\n';
}
```
That returns like:
```
1
4294967296
```
(first number is implementation dependent, but 32 bit, second is 64 bit)